File: Script/Tools/SceneEditor/Panels/ConsolePanel.lua
-- [ts]: ConsolePanel.ts
local ____exports = {} -- 1
local ImGui = require("ImGui") -- 1
local ____Theme = require("Script.Tools.SceneEditor.Theme") -- 3
local okColor = ____Theme.okColor -- 3
local themeColor = ____Theme.themeColor -- 3
local ____Model = require("Script.Tools.SceneEditor.Model") -- 4
local zh = ____Model.zh -- 4
function ____exports.drawConsolePanel(state) -- 6
	ImGui.TextColored(themeColor, zh and "控制台" or "Console") -- 7
	ImGui.SameLine() -- 8
	ImGui.TextColored(okColor, state.status) -- 9
	ImGui.Separator() -- 10
	for ____, line in ipairs(state.console) do -- 11
		ImGui.TextDisabled(line) -- 11
	end -- 11
end -- 6
return ____exports -- 6

File: Script/Tools/SceneEditor/Panels/ConsolePanel.ts
import * as ImGui from 'ImGui';
import { EditorState } from 'Script/Tools/SceneEditor/EditorTypes';
import { okColor, themeColor } from 'Script/Tools/SceneEditor/Theme';
import { zh } from 'Script/Tools/SceneEditor/Model';

export function drawConsolePanel(state: EditorState) {
	ImGui.TextColored(themeColor, zh ? '控制台' : 'Console');
	ImGui.SameLine();
	ImGui.TextColored(okColor, state.status);
	ImGui.Separator();
	for (const line of state.console) ImGui.TextDisabled(line);
}
